JoonasHeinonen/escape-from-zordoom-prison
# Feature/vendor system

### [Vendor System](https://trello.com/c/Kne8H4YK/25-vendor-system)

![image](https://user-images.githubusercontent.com/33086486/209235392-2ccee67f-5b81-41e8-bac7-89bda2d09103.png)

**Changes:**
 - Imported Slim Cognito's 3D model and vendor images for the weapons.
 - Programmed the logic for the vendor system. If not enough bolts, you ain't buying a thing. If you're enough... Mmmmm... Rich, then you can buy something.
 - Designed the UI for the vendor system.
 - Changed the position of the boltCounter from `test_level.tscn` to `player.tscn`.

**How to test:**
 - See that you can purchase anything from the vendor.
 - Verify that all the arrays function. (This was one of the main issues in implementing the logic of the vendor system.)
 - Verify that the code is according to good GD standards.

**Sources cited:**
 - [Removing items from the middle of an array
](https://www.reddit.com/r/godot/comments/ggya8f/removing_items_from_the_middle_of_an_array/)
 - [Remove all children?](https://www.reddit.com/r/godot/comments/9qmjfj/remove_all_children/)
 - [Removing an object from an array](https://godotengine.org/qa/18239/removing-an-object-from-an-array)
